trinity/config.py

```diff
@@ -457,14 +500,13 @@ def node_class(self) -> Type['Node']:
         from trinity.nodes.full import FullNode
         from trinity.nodes.light import LightNode
 
-        if self.is_full_mode:
+        if self.base_config.is_full_mode:
             return FullNode
-        elif self.is_light_mode:
```

Comment:
I'm not sure if the sync modes itself should also move into the `TrinityEth1SubConfig`. They probably should? Thoughts?

Reply:
I think they should be on the `Eth1SubConfig`

Reply:
Ok, cool. Will move that once we've sorted the core constructor/from_parser_args issue out.